src/sage/schemes/riemann_surfaces/riemann_surface.py

```diff
@@ -3990,6 +4013,806 @@ def vectomat(v):
+                    break
+            if pivot:
+                break
+        else:
+            break
+
+        i, j = pivot
+
+        # Move pivot to (t, t+1)
+        if i != t:
+            P = Matrix.identity(ZZ, total_size)
+            P.swap_columns(i, t)
+            M_curr = P.transpose() * M_curr * P
+            S = S * P
+
+        if j != t + 1:
+            P = Matrix.identity(ZZ, total_size)
+            P.swap_columns(j, t + 1)
+            M_curr = P.transpose() * M_curr * P
+            S = S * P
+
+        # ensure pivot entry is positive
+        if M_curr[t, t + 1] < 0:
+            P = Matrix.identity(ZZ, total_size)
+            P[t + 1, t + 1] = -1
+            M_curr = P.transpose() * M_curr * P
+            S = S * P
+
+        # clear other entries in rows/cols t and t+1
+        for r in range(total_size):
+            if r in (t, t + 1):
+                continue
+
+            # zero M[t, r] using the pair (a, d) with d = M[t, t+1]
+            a = ZZ(M_curr[t, r])
+            d = ZZ(M_curr[t, t + 1])
+            if a != 0 or d != 0:
+                if a == 0 and d == 0:
+                    pass
+                else:
+                    g1, u, v = xgcd(ZZ(a), ZZ(d))
+                    p = ZZ(d) // g1
+                    r_coeff = -ZZ(a) // g1
+                    q = ZZ(u)
+                    s = ZZ(v)
+                    U1 = Matrix(ZZ, [[p, q], [r_coeff, s]])
+
+                    E1 = Matrix.identity(ZZ, total_size)
+                    # place U1 into rows/cols (r, t+1)
+                    E1[r, r] = U1[0, 0]
+                    E1[r, t + 1] = U1[0, 1]
+                    E1[t + 1, r] = U1[1, 0]
+                    E1[t + 1, t + 1] = U1[1, 1]
+
+                    M_curr = E1.transpose() * M_curr * E1
+                    S = S * E1
+
+            # zero M[t+1, r] using the pair (b, -d)
+            b = ZZ(M_curr[t + 1, r])
+            d = ZZ(M_curr[t, t + 1])
+            if b != 0 or d != 0:
+                if b == 0 and (-d) == 0:
+                    pass
+                else:
+                    g2, u, v = xgcd(ZZ(b), ZZ(-d))
+                    p = ZZ(-d) // g2
+                    r_coeff = -ZZ(b) // g2
+                    q = ZZ(u)
+                    s = ZZ(v)
+                    U2 = Matrix(ZZ, [[p, q], [r_coeff, s]])
+
+                    E2 = Matrix.identity(ZZ, total_size)
+                    E2[r, r] = U2[0, 0]
+                    E2[r, t] = U2[0, 1]
+                    E2[t, r] = U2[1, 0]
+                    E2[t, t] = U2[1, 1]
+
+                    M_curr = E2.transpose() * M_curr * E2
+                    S = S * E2
+
+        # ensure positivity of the 2x2 pivot
+        if M_curr[t, t + 1] < 0:
+            P = Matrix.identity(ZZ, total_size)
+            P[t + 1, t + 1] = -1
+            M_curr = P.transpose() * M_curr * P
+            S = S * P
+
+        t += 2
+
+    # enforce divisibility of adjacent blocks
+    if pair_count >= 2:
```

Comment:
because the following `for` loop is over `range(pair_count - 1)` then if `pair_count == 1` the for loop is over an empty iterator. As such, you do not need the `if pair_count >= 2` condition, and that reduces the indentation of this code.